frontend/src/pages/NewAgreement.tsx

```diff
@@ -66,14 +67,15 @@ const NewAgreement: FC = () => {
 
   async function handleContinueClick() {
     if (step === 3) {
-      await addAgreement(categoryId, agreementName, rationale, chapters);
+      const agreementData = await addAgreement(categoryId, agreementName, rationale, chapters);
+      setAgreementId(agreementData.data.insert_core_agreements_one.name);
```

Comment:
Oops! did you mean `insert_core_agreements_one.id`?

Reply:
no, the name is missleading. this is the name, not id. fixed

Reply:
yes, fixed